### ISSUE TITLE
Brackets that come under each other should not pass

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#385]
# Bug report

## What's wrong

Currently we allow brackets that located like so:
```python
xy = [[
]
]
```

## How is that should be

This should raise a violation.
Correct code samples:

```python
xy = [[]]
xy = [
    [],
]
xy = [[
]]
```

**Comment by sobolevn** [comment]
It raises `tests/fixtures/noqa.py:2:2: C812 missing trailing comma` which is good enough.

And this example:

```python
xy = [[
],
]
```

still should not be allowed.

**Comment by sobolevn** [comment]
This will be marked as a low priority.
Since the algo to count parens like this is too complex. And the profit is not so big.

**Comment by sobolevn** [comment]
I cannot fight these corner cases:

```python
x = [
    [1],   # note [ position
    [2],  # note [ position
]
```

This is a valid case. But, `[` comes just under `[`
So, it is hard to tell the difference between these two cases.

**Comment by sobolevn** [comment]
To fix this we also need to count "level" of brackets. It is possible to nest brackets of the same level. But, not different levels.
